packages/modular-scripts/src/utils/getPackageDependencies.ts

```diff
@@ -109,30 +114,37 @@ function parseYarnLock(lockFile: string, deps: Dependency): Dependency {
 }
 
 function parseYarnLockV1(
-  lockFile: string,
+  lockFileContents: string,
   deps: Dependency,
 ): Dependency | null {
+  let parsedLockfile: ReturnType<typeof lockfile.parse>;
   try {
-    const parsedLockfile = lockfile.parse(lockFile);
-    return Object.entries(deps).reduce<Record<string, string>>(
-      (acc, [name, version]) => {
-        acc[name] = (parsedLockfile.object as LockFileEntries)[
-          `${name}@${version}`
-        ].version;
-        return acc;
-      },
-      {},
-    );
+    parsedLockfile = lockfile.parse(lockFileContents);
   } catch (e) {
     return null;
   }
+  return Object.entries(deps).reduce<Record<string, string>>(
+    (acc, [name, version]) => {
+      const resolution = (parsedLockfile.object as LockFileEntries)[
+        `${name}@${version}`
+      ]?.version;
+      if (resolution !== undefined) {
+        acc[name] = resolution;
+      }
+      return acc;
+    },
+    {},
+  );
```

Comment:
As the inline type assertion splits this over several lines, you could flatten this by making the assertion outside of the `.reduce`:

```typescript
const lockFileEntries = parsedLockfile.object as LockFileEntries;
  return Object.entries(deps).reduce<Record<string, string>>(
    (acc, [name, version]) => {
      const key = `${name}@${version}`;
      if (lockFileEntries[key]) {
        acc[name] = lockFileEntries[key].version;
      }
      return acc;
    },
    {},
  );
```

Reply:
👍 done